misk/src/test/kotlin/misk/web/WebDispatchTest.kt

```diff
@@ -0,0 +1,134 @@
+package misk.web
+
+import com.google.common.truth.Truth.assertThat
+import com.google.common.util.concurrent.ServiceManager
+import com.google.inject.Guice
+import com.google.inject.Provides
+import com.squareup.moshi.Moshi
+import misk.MiskModule
+import misk.inject.KAbstractModule
+import misk.inject.uninject
+import misk.testing.PortPicker
+import misk.web.actions.WebAction
+import okhttp3.HttpUrl
+import okhttp3.MediaType
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+import javax.inject.Singleton
+
+internal class WebDispatchTest {
+    data class HelloBye(val message: String)
+
+    val jsonMediaType = MediaType.parse("application/json")
+
+    @Inject lateinit var serviceManager: ServiceManager
+    @Inject lateinit var webConfig: WebConfig
+    @Inject lateinit var moshi: Moshi
+
+    @Before
+    fun before() {
+        uninject(this)
+
+        val injector = Guice.createInjector(
+                WebModule(),
+                TestModule(),
+                MiskModule()
+        )
+
+        injector.injectMembers(this)
+        serviceManager.startAsync()
+        serviceManager.awaitHealthy(5, TimeUnit.SECONDS)
+    }
+
+    @After
+    fun after() {
+        serviceManager.stopAsync()
+        serviceManager.awaitStopped(5, TimeUnit.SECONDS)
+    }
+
+    @Test
+    fun post() {
+        val requestContent = helloByeJsonAdapter.toJson(HelloBye("my friend"))
+        val httpClient = OkHttpClient()
+        val request = Request.Builder()
+                .post(RequestBody.create(jsonMediaType, requestContent))
+                .url("${serverUrl}hello")
+                .build()
+
+        val response = httpClient.newCall(request).execute()
+        assertThat(response.code()).isEqualTo(200)
+        val responseContent  = response.body()!!.source()
+        assertThat(helloByeJsonAdapter.fromJson(responseContent)!!.message).isEqualTo("post hello my friend")
```

Comment:
Wanna actually assert exactly the exact JSON? Moshi is deterministic (whitespace etc.) and then there’s one fewer things in the test infrastructure.

Reply:
(and likewise for the request body)